image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -141,7 +141,18 @@ class ImageLoaderController(auth: Authentication,
 
         val approximateReceiveCount = getApproximateReceiveCount(sqsMessage)
 
-        if (approximateReceiveCount > 2) {
+        if(config.maybeUploadLimitInBytes.exists(_ < s3IngestObject.contentLength)){
+          val errorMessage = s"File size exceeds the maximum allowed size (${config.maybeUploadLimitInBytes.get / 1_000_000}MB). Moving to fail bucket."
+          logger.warn(logMarker, errorMessage)
+          store.moveObjectToFailedBucket(s3IngestObject.key)
+          s3IngestObject.maybeMediaIdFromUiUpload foreach { imageId =>
+            uploadStatusTable.updateStatus( // fire & forget, since there's nothing else we can do
+              imageId, UploadStatus(StatusType.Failed, Some(errorMessage))
+            )
+          }
+          Future.unit
```

Comment:
As far as I can tell this code path doesn't set off an increment in the `failedIngestsFromQueue` metric, which is incremented when an upload fails due to the wrong mimetype, for instance. Should we be incrementing the metric for size-related failures too?

Reply:
nice spot, added in https://github.com/guardian/grid/pull/4387/commits/064daab40a8782c6f0dd08e27f31945614d242ff